### PR TITLE
Api 34330 split controllers

### PR DESCRIPTION
**Pull request by acovrig**
## Summary
- Split v2 POA controller into individual vs organization

## Related issue(s)
- [API-34330](https://jira.devops.va.gov/browse/API-34330)

## Testing done
- [x] *New code is covered by unit tests*

## What areas of the site does it impact?
Mostly QoL improvement for devs by splitting the V2 POA controller to 2 separate files.

## Acceptance criteria
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/individual_controller.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+        def validate_individual_poa_code!(poa_code)
+          return if ::Veteran::Service::Representative.where('? = ANY(poa_codes)', poa_code).any?
+
+          raise ::ClaimsApi::Common::Exceptions::Lighthouse::ResourceNotFound.new(
+            detail: "Could not find an Accredited Representative with code: #{poa_code}"
+          )
+        end
+
+        def parse_and_validate_poa_code(form_number)
```
Comment (jvcAdHoc) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/individual_controller.rb:40]: I can't find uses of this method. Is it legacy from the original v2 implementation?

File: modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/organization_controller.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+        def validate_org_poa_code!(poa_code)
+          return if ::Veteran::Service::Organization.exists?(poa: poa_code)
+
+          raise ::ClaimsApi::Common::Exceptions::Lighthouse::ResourceNotFound.new(
+            detail: "Could not find an Organization with code: #{poa_code}"
+          )
+        end
+
+        def current_poa
```
Comment (jvcAdHoc) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/organization_controller.rb:48]: This is also defined on the base class

File: modules/claims_api/config/routes.rb [hunk 3]
```diff
@@ -43,14 +43,16 @@
...
-      post '/:veteranId/power-of-attorney:appoint-individual', to: 'power_of_attorney#appoint_individual',
-                                                               constraints: { 'appoint-individual': /:appoint-individual/ } # rubocop:disable Layout/LineLength
+      scope module: 'power_of_attorney', path: '' do
+        get '/:veteranId/power-of-attorney', to: 'base#show'
+        post '/:veteranId/2122/validate', to: 'organization#validate2122'
+        post '/:veteranId/2122', to: 'organization#submit2122'
+        post '/:veteranId/2122a', to: 'individual#submit2122a'
+        post '/:veteranId/2122a/validate', to: 'individual#validate2122a'
+        get '/:veteranId/power-of-attorney/:id', to: 'base#status'
+        post '/:veteranId/power-of-attorney:appoint-individual', to: 'individual#appoint_individual',
```
Comment (jvcAdHoc) [inline modules/claims_api/config/routes.rb:53]: I think this route can be removed

**Comment by jvcAdHoc** [comment]
Does it make sense to also split the request spec into org + individual?

**Comment by acovrig** [comment]
> Does it make sense to also split the request spec into org + individual?

A part of me thinks that this should be a separate PR, but I already hit the dangerbot w/out it so eh

File: modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb [hunk 4]
```diff
@@ -113,14 +100,6 @@ def representative(poa_code)
           format_representative(individuals.first)
         end
 
-        def format_organization(organization)
```
Comment (jvcAdHoc) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb:116]: The base class needs this for the show endpoint

File: modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/individual_controller.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    module Veterans
+      class PowerOfAttorney::IndividualController < ClaimsApi::V2::Veterans::PowerOfAttorney::BaseController
+        FORM_NUMBER = '2122A'
+
+        def submit
+          shared_form_validation(FORM_NUMBER)
+          poa_code = get_poa_code(FORM_NUMBER)
+          validate_individual_poa_code!(poa_code)
+
+          submit_power_of_attorney(poa_code, FORM_NUMBER)
```
Comment (stiehlrod) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/individual_controller.rb:19]: Screenshot 2024-03-05 at 11 09 51 AM

**Review by jvcAdHoc — APPROVED** [review]

**Review by ryan-mcneil — APPROVED** [review]
:shipit:

**Pull request merged**